src/app/course-details/course-details.component.ts

```diff
@@ -13,25 +17,34 @@ export class CourseDetailsComponent implements OnInit{
 
   courseItems: CourseItem[] = [];
 
-  constructor(public courseService: CourseServiceService) {}
+  constructor(public courseService: CourseServiceService,private spinner: NgxSpinnerService) {}
 
   ngOnInit() {
     this.loadCourses();
   }
 
   loadCourses(){
+    this.spinner.show();
     this.courseService.getAllCourses().subscribe(courseWrapper => {
```

Comment:
Better to use rxjs operator `catchError` inside `pipe`, instead of error callback in `subscribe`.